File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/Appointment.cs
namespace CareFirstClinic.API.Models
{
    public enum AppointmentStatus
    {
        Pending,
        Confirmed,
        Waiting,
        InProgress,
        Completed,
        Cancelled
    }

    public class Appointment
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid PatientId { get; set; }
        public Guid TimeSlotId { get; set; }

        public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;

        public bool IsConsultationPaid { get; set; } = false;
        public bool IsMedicinePaid { get; set; } = false;

        public decimal ConsultationFee { get; set; } = 0;
        public decimal MedicineFee { get; set; } = 0;

        public string? ServiceName { get; set; }

        public string? Reason { get; set; }
        public string? CancelReason { get; set; }
        public DateTime? CancelledAt { get; set; }
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation
        public Patient? Patient { get; set; }
        public TimeSlot? TimeSlot { get; set; }
        public MedicalRecord? MedicalRecord { get; set; }
        public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    }
}

File: src/frontend/src/types/appointment.ts
export interface Appointment {
  id: string;
  patientId: string;
  patientName: string;
  timeSlotId: string;
  doctorId: string;
  doctorName: string;
  specialtyName: string;
  serviceName?: string;
  consultationFee: number;
  medicineFee: number;
  isConsultationPaid: boolean;
  isMedicinePaid: boolean;
  // Backward-compatible fields used by older UI components
  totalAmount?: number;
  paymentStatus?: 'Unpaid' | 'Paid' | 'Refunded';
  workDate: string;
  startTime: string;
  endTime: string;
  status: 'Pending' | 'Confirmed' | 'Waiting' | 'InProgress' | 'Completed' | 'Cancelled';
  reason?: string;
  cancelReason?: string;
  cancelledAt?: string;
  notes?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateAppointmentDTO {
  timeSlotId: string;
  reason?: string;
  notes?: string;
  fullName: string;
  dob: string;
  gender: string;
  phone: string;
  email?: string;
  serviceName?: string;
  consultationFee: number;
}

export interface UpdateAppointmentDTO {
  reason?: string;
  notes?: string;
  status?: 'Pending' | 'Confirmed' | 'Waiting' | 'InProgress' | 'Completed' | 'Cancelled';
  paymentStatus?: 'Unpaid' | 'Paid' | 'Refunded';
}

export interface CancelAppointmentDTO {
  cancelReason: string;
}

export interface AppointmentQueryParams {
  patientId?: string;
  doctorId?: string;
  today?: boolean;
  status?: string;
  paymentStatus?: 'Unpaid' | 'Paid' | 'Refunded';
  page?: number;
  pageSize?: number;
  fromDate?: string;
  toDate?: string;
  sortBy?: string;
  sortDir?: string;
  search?: string;
}